README.md

```diff
@@ -1,21 +1,97 @@
-### Assigning students to courses
+# Application - Assigning students to courses
 
-This program is a simple app that stores courses and students data and assigns students to a course.
+This program is a simple app that stores courses and students data and can assigns students to a course. It is a simple REST API written in python using Flask.
 
-IT is a simple REST API written in python using Flask.
+## REST API
 
-## What is REST API and Flask
+A REST (REpresentational State Transfer) API (Application Programming Interface) is a transfer of resources (data) from a Server to a Client over the internet. Through the REST API, the client can perform a GET, POST, PUT, DELETE request to the server.
 
-> what api provides to client
+A client uses an API to request for resources from the server. A server has the resources and transfers it to the client through an API. A resource can be any object for example, user, photo, number. An API allows one software to talk to another. It is a procedure allowing a client to access the data from a server. 
 
+An example of an API is "https://github.com/StephenDsouza90/demo-crud-sqlite-python-app" where:
+ 
+github.com/StephenDsouza90 - Endpoint URL
 
-##  The set up
+demo-crud-sqlite-python-app - Parameter
```

Comment:
`github.com` is the endpoint

`StephenDsouza90/demo-crud-sqlite-python-app` is the resource in which `/StephenDsouza90` is the parameter for the github username and `/demo-crud-sqlite-python-app` is the parameter for the github user's project name.

Reply:
change made to endpoint